src/Fist/Http/tests/RequestTest.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+namespace FistTest;
+
+use Fist\Http\Request;
+use Fist\Testing\TestCase;
+
+class RequestTest extends TestCase
+{
+    public function testBuildingInstance()
+    {
+        $request = new Request();
+
+        $this->assertInstanceOf(Request::class, $request);
+    }
+
+    public function testBuildingFromGlobals()
+    {
+        $request = Request::createFromGlobals();
+
+        $this->assertInstanceOf(Request::class, $request);
+    }
+
+    public function testGettingRequestMethod()
+    {
+        $request = Request::createFromGlobals();
+
+        $this->assertEquals('GET', $request->getMethod());
+    }
+
+    public function testGettingFakedRequestMethod()
+    {
+        $_SERVER['REQUEST_METHOD'] = 'POST';
+
+        $request = Request::createFromGlobals();
+        unset($_SERVER['REQUEST_METHOD']);
+
+        $this->assertEquals('POST', $request->getMethod());
+    }
+
+    public function testOverwritingRequestMethod()
+    {
+        $request = Request::createFromGlobals();
+
+        $this->assertEquals('GET', $request->getMethod());
+
+        $request->setMethod('PATCH');
+
+        $this->assertEquals('PATCH', $request->getMethod());
+    }
+
+    public function testGettingRequestUrl()
```

Comment:
__[phpmd]__
* testGettingRequestUrl accesses the super-global variable $_SERVER.

<!-- Comment by [SideCI](https://sideci.com) -->